### DISCUSSION TITLE
let mut, var imm or let var, var let

### DISCUSSION THREAD

**Discussion by igotfr** [carbon-language/carbon-lang#1669] [Language design]
For instance:
```rust
let (a, mut b, c) = (d, e, f);
```
```rust
var (a, imm b, c) = (d, e, f);
```
or
```rust
let (a, var b, c) = (d, e, f);
```
```rust
var (a, let b, c) = (d, e, f);
```

**Comment by lexi-nadia** [comment]
This could be written using current syntax as:

```swift
let a = d;
var b = e;
let d = f;
```

Personally, i find this easier to read than the proposal; i don't have to correlate names on the left with values on the right.

Do you have a motivating example to show how the proposed syntax would help?

**Reply by igotfr** [reply]
> This could be written using current syntax as:
> 
> ```swift
> let a = d;
> var b = e;
> let d = f;
> ```
> 
> Personally, i find this easier to read than the proposal; i don't have to correlate names on the left with values on the right.
> 
> Do you have a motivating example to show how the proposed syntax would help?

I can want destructuring a tuple, array or struct which each member has different mutabilities

**Reply by jonmeow** [reply]
At present, I think that's

```
let (a: auto, var b: auto, c: auto) = (d, e, f);
```

To try to summarize, this is because `var` is supposed to guarantee storage, whereas `let` isn't. So a `var` inside `let` is essentially saying "even though the tuple doesn't need storage, this part does". By contrast, it's hard to say "even though this tuple needs storage, this part doesn't".

Example: https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/testdata/let/nested_tuple_pattern.carbon
